code/modules/atmospherics/machinery/components/binary_devices/thermomachine.dm

```diff
@@ -300,6 +340,27 @@
 	update_appearance()
 	return TRUE
 
+/obj/machinery/atmospherics/components/binary/thermomachine/proc/check_explosion(temperature)
+	if(temperature < FUSION_MAXIMUM_TEMPERATURE + 2000)
+		return FALSE
+	if(prob(log(100000000, temperature) * 10)) //10% at 1e8
+		return TRUE
```

Comment:
How quickly could temp escalate/is there a hard cap to this?

Reply:
The temp can rise pretty fast if there is a big source of heat, so the explosion could be very fast and instant. But I don't think it is an issue because to make the heat grow that much you'd need to have very few moles of gas on the receiving side with low heat capacity. And in any cases you'd need to remove the safeties from the machine